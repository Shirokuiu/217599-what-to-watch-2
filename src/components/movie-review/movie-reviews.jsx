import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import Reviews from "../reviews/reviews";

import {Operation} from "../../reducer";

export const MovieReviews = (props) => {
  const {onLoadComments, movieId} = props;

  onLoadComments(movieId);

  return <div className="movie-card__reviews movie-card__row">
    <Reviews />
  </div>;
};

MovieReviews.propTypes = {
  movieId: PropTypes.number.isRequired,
  onLoadComments: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  onLoadComments: (movieId) => {
    dispatch(Operation.loadComments(movieId));
  }
});

export default connect(null, mapDispatchToProps)(MovieReviews);
