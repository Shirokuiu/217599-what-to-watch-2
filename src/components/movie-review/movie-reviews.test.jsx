import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import {MovieReviews} from "./movie-reviews";

Enzyme.configure({adapter: new Adapter()});

it(`Is movie review rendered`, () => {
  const tree = shallow(<MovieReviews
    movieId={1}
    onLoadComments={jest.fn()}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
