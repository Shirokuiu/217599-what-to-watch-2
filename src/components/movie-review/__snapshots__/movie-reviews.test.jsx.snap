// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is movie review rendered 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <Connect(Reviews) />
</div>
`;
